### PR TITLE
Upgrade soto to 7 and enable Swift 6 language mode

### PR DESCRIPTION
**Pull request by giginet**
I updated soto to 7.x

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -28,15 +28,13 @@ let package = Package(
...
             plugins: [
                 .plugin(name: "SotoCodeGeneratorPlugin", package: "soto-codegenerator"),
             ]
         ),
         .testTarget(
             name: "ScipioS3StorageTests",
             dependencies: ["ScipioS3Storage"]),
-    ]
+    ],
+    swiftLanguageModes: [.v6]
```
Comment (mtj0928) [inline Package.swift:39]: This is not required, because the default is `.v6`.

File: Tests/ScipioS3StorageTests/CompressorTests.swift [hunk 2]
```diff
@@ -41,9 +37,11 @@ final class CompressorTests: XCTestCase {
             String(bytes: fileContents, encoding: .utf8),
             fileBody
         )
+    }
+
+    override func tearDown() {
+        super.setUp()
```
Comment (mtj0928) [inline Tests/ScipioS3StorageTests/CompressorTests.swift]:
```suggestion
        super.tearDown()
```

File: Tests/ScipioS3StorageTests/CompressorTests.swift [hunk 3]
```diff
@@ -41,9 +37,11 @@ final class CompressorTests: XCTestCase {
...
 
-        addTeardownBlock {
-            try? self.fileManager.removeItem(at: xcframeworkPath)
```
Comment (giginet) [inline Tests/ScipioS3StorageTests/CompressorTests.swift:46]: `xcframeworkPath` is under `workspacePath`

File: Package.swift (hunk 1 above)
Comment (giginet) [reply Package.swift:39]: I prefer to declare this explicitly

**Review by mtj0928 — APPROVED** [review]

**Pull request merged**